### DISCUSSION TITLE
How should we call `Package` class in deeplinks and should we add its instance in deeplinks?

### DISCUSSION THREAD

**Discussion by FreePhoenix888** [deep-foundation/Discussions#30] [Ideas]
We have [this class](https://github.com/deep-foundation/deeplinks/blob/662002adc587adcc78f5311b50199bd6d722f260/imports/package.ts)
Should we call it `BaseClass`?
Why do I have this question? Because I would like core package to have its `Package` class too as other packages will have.
As I remember core package is not fully separated from deeplinks right now

**Comment by konard** [comment]
@FreePhoenix888 we may keep `Package` name it does not conflicts with other names. Core package with have CorePackage class that extends Package. And the core package will be fully separate soon.

**Reply by FreePhoenix888** [reply]
So you think we should add name to Package classes in every package? Imagine in it firebase push notification package 🤣
I think we can name them Package. There are modules and aliases for these reasons In js

**Reply by konard** [reply]
Yes, there will be different types of packages. Some packages can import multiple over types, so it is a good idea to name them. Or the user will have to do it anyway in that case they will have find a way to name them (I mean make up new name each time). We can name these packages beforehand so users will just use existing prepared name.
